### DISCUSSION TITLE
What is the best way to trace GRPC stream messages in ASP.NET Core?

### DISCUSSION THREAD

**Discussion by dceravigupta** [open-telemetry/opentelemetry-dotnet#4722] [General]
I have a gRPC based ASP.NET Core service which expose streaming gRPC APIs. Currently I'm using OpenTelemetry SDK auto collection to collect traces for all incoming calls automatically but seems like it doesn't capture traces for individual request in case of a streaming gRPC API. Any recommendation how we can auto collect them?

![image](https://github.com/open-telemetry/opentelemetry-dotnet/assets/32182026/69719fea-e277-45b3-b125-e19a49b0c693)

**Comment by philliphoff** [comment]
@dceravigupta I've noticed the same; I see traces for unary methods but none for streaming methods.  Were you able to resolve this?

**Comment by dceravigupta** [comment]
No
